swiftsimio/writer.py

```diff
@@ -200,6 +201,69 @@ def write_particle_group(self, file_handle: h5py.File, compress: bool):
 
         return
 
+    def write_particle_group_metadata(self, file_handle: h5py.File, dset_attributes):
+        for name, output_handle in getattr(
+            metadata.required_fields, self.particle_name
+        ).items():
+            obj = file_handle[f"/PartType{self.particle_type}/{output_handle}"]
+            for attr_name, attr_value in dset_attributes[output_handle].items():
+                obj.attrs.create(attr_name, attr_value)
+
+        return
+
+    def get_attributes(self, scale_factor):
+        attributes_dict = {}
+        a_exp_dict = {
+            "coordinates": 1,
+            "internal_energies": -2,
+        }
+
+        for name, output_handle in getattr(
+            metadata.required_fields, self.particle_name
+        ).items():
+            field = getattr(self, name)
+
+            # Find the exponents for each of the dimensions
+            dim_exponents = get_dimensions(field.units.dimensions)
+
+            # Find the scale factor associated quantities
+            a_exp = a_exp_dict.get(name, 0)
+            a_factor = scale_factor * a_exp
+
+            attributes_dict[output_handle] = {
+                "Conversion factor to CGS (not including cosmological corrections)": [
+                    field.unit_quantity.in_cgs()
+                ],
+                "Conversion factor to physical CGS (including cosmological corrections)": [
+                    field.unit_quantity.in_cgs() * a_factor
+                ],
+                "Description": b"Co-moving positions of the particles",
+                "Expression for physical CGS units": b"a U_L  [ cm ]",
+                "U_I exponent": [dim_exponents[0]],
+                "U_L exponent": [dim_exponents[1]],
+                "U_M exponent": [dim_exponents[2]],
+                "U_T exponent": [dim_exponents[3]],
+                "U_t exponent": [dim_exponents[4]],
+                "a-scale exponent": [a_exp],
+                "h-scale exponent": [0.0],
+            }
+
+        return attributes_dict
+
+
+def get_dimensions(obj):
+    dimensions = ["current", "length", "mass", "temperature", "time"]
```

Comment:
Is this list not present anywhere in unyt?

Reply:
There is indeed. I changed it to use the one provided in unyt. unfortunately a bit of hacking needed to happen because they use current_mks instead of current.